firefox-ios/Client/Frontend/Browser/TermsOfUse/TermsOfUseTelemetry.swift

```diff
@@ -2,28 +2,96 @@
     }
 
-    func termsOfUseAcceptButtonTapped() {
+    func termsOfUseAcceptButtonTapped(surface: Surface = .bottomSheet, acceptedDate: Date) {
         let acceptedExtra = GleanMetrics.Termsofuse.AcceptedExtra(
-            surface: termsOfUseSurface,
+            surface: surface.rawValue,
+            touVersion: String(termsOfUseVersion)
+        )
+        gleanWrapper.recordEvent(for: GleanMetrics.Termsofuse.accepted, extras: acceptedExtra)
+        gleanWrapper.recordQuantity(for: GleanMetrics.Termsofuse.version, value: termsOfUseVersion)
+        gleanWrapper.recordDatetime(for: GleanMetrics.Termsofuse.date, value: acceptedDate)
+    }
+
+    func termsOfUseRemindMeLaterButtonTapped(surface: Surface = .bottomSheet) {
+        let remindMeLaterExtra = GleanMetrics.Termsofuse.RemindMeLaterClickExtra(
+            surface: surface.rawValue,
+            touVersion: String(termsOfUseVersion)
+        )
+        gleanWrapper.recordEvent(for: GleanMetrics.Termsofuse.remindMeLaterClick, extras: remindMeLaterExtra)
+        gleanWrapper.incrementCounter(for: GleanMetrics.Termsofuse.remindMeLaterCount)
+    }
+
+    func termsOfUseLearnMoreButtonTapped(surface: Surface = .bottomSheet) {
+        let learnMoreExtra = GleanMetrics.Termsofuse.LearnMoreClickExtra(
+            surface: surface.rawValue,
+            touVersion: String(termsOfUseVersion)
+        )
+        gleanWrapper.recordEvent(for: GleanMetrics.Termsofuse.learnMoreClick, extras: learnMoreExtra)
+    }
+
+    func termsOfUsePrivacyNoticeLinkTapped(surface: Surface = .bottomSheet) {
+        let privacyNoticeExtra = GleanMetrics.Termsofuse.PrivacyNoticeClickExtra(
+            surface: surface.rawValue,
+            touVersion: String(termsOfUseVersion)
+        )
+        gleanWrapper.recordEvent(for: GleanMetrics.Termsofuse.privacyNoticeClick, extras: privacyNoticeExtra)
+    }
+
+    func termsOfUseTermsOfUseLinkTapped(surface: Surface = .bottomSheet) {
+        let termsOfUseExtra = GleanMetrics.Termsofuse.TermsOfUseClickExtra(
+            surface: surface.rawValue,
             touVersion: String(termsOfUseVersion)
         )
-        GleanMetrics.Termsofuse.accepted.record(acceptedExtra)
-        GleanMetrics.Termsofuse.version.set(termsOfUseVersion)
-        GleanMetrics.Termsofuse.date.set(Date())
+        gleanWrapper.recordEvent(for: GleanMetrics.Termsofuse.termsOfUseClick, extras: termsOfUseExtra)
+    }
+
+    func termsOfUseDismissed(surface: Surface = .bottomSheet) {
+        let dismissExtra = GleanMetrics.Termsofuse.DismissExtra(
+            surface: surface.rawValue,
+            touVersion: String(termsOfUseVersion)
+        )
+        gleanWrapper.recordEvent(for: GleanMetrics.Termsofuse.dismiss, extras: dismissExtra)
+        gleanWrapper.incrementCounter(for: GleanMetrics.Termsofuse.dismissCount)
+    }
+
+    static func setUsageMetrics(gleanWrapper: GleanWrapper = DefaultGleanWrapper()) {
```

Comment:
I think it would be best to add the profile as a parameter with default value as well for easier testabiliy.

Reply:
Yes, I added it as a parameter with default value. Thank you!